Src/BootCamp.Chapter/MacFactory.cs

```diff
@@ -5,11 +5,58 @@
 
 namespace BootCamp.Chapter
 {
-    public class MacFactory
+    public class MacFactory : Factory
     {
-        public DesktopComputer Assemble()
+        private const string manufacturer = "Apple Inc.";
+
+        public MacFactory() { }
+
+        protected override void PrepareBody(DesktopComputer computer)
+        {
+            Body body = new Body("MC-PrC-500", "Apple", "Mac Pro Case 500", CaseDesign.ATX);
+            Console.WriteLine($"Preparing body with ID: {body.GetID()}, and ModelName: {body.GetModelName()} for assembly.");
+            computer.SetBody(body);
+        }
+
+        protected override void InstallMotherboard(DesktopComputer computer)
+        {
+            Motherboard motherboard = new Motherboard("ARS-Z390-E-G", "Asus", "ASUS ROG STRIX Z390-E GAMING", "Z390-E", "Intel", SocketType.LGA1151_300);
+            Console.WriteLine($"Installing motherboard {motherboard.GetModelName()} onto motherboard.");
+            computer.SetMotherboard(motherboard);
+        }
+
+        protected override void InstallCpu(DesktopComputer computer)
+        {
+            Cpu cpu = new Cpu("INTEL-I5-9600K", "Intel", "INTEL Core i5-9600K", SocketType.LGA1151_300, 6, 6, 3700);
+            Console.WriteLine($"Applying thermal paste to cpu socket... and installing {cpu.GetModelName()} onto motherboard.");
```

Comment:
Yes, I know you apply it to the CPU not the socket, but hey, it's Apple right? 😄